### PR TITLE
Add additional LineStyles

### PR DESCRIPTION
**Pull request by Nutzzz**
Added dashed, dotted, and thick line styles and combinations where available.

## Pull Request checklist:

- [X] I've named my PR in the form of "Fixes #issue. Terse description."
- [X] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [X] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [X] I ran `dotnet test` before commit
- [X] I have made corresponding changes to the API documentation (using `///` style comments)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any poor grammar or misspellings
- [X] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by Nutzzz** [comment]
I added a few Unit Tests dealing with thick and thin combinations, but I'll make more for the dashed and dotted variations.

**Comment by tig** [comment]
This is fantastic!

Note that I'm knees deep in a massive refactoring of related border LineCanvas stuff right now that will greatly benefit from your work. 

However, this also means I'm going to hold off reviewing and merging this pr until I'm done. 

If you want, you can retarget this pr to be a pr to my pr (#2527) which may streamline things.

File: Terminal.Gui/Drawing/LineCanvas.cs [hunk 1]
```diff
@@ -249,99 +277,118 @@ private abstract class IntersectionRuneResolver {
...
 
 				if (doubleHorizontal) {
 					return doubleVertical ? doubleBoth : doubleH;
 				}
-
 				if (doubleVertical) {
 					return doubleV;
 				}
 
+				if (thickHorizontal) {
```
Comment (tig) [inline Terminal.Gui/Drawing/LineCanvas.cs:319]: I think there's  bug in here somewhere:

image

Comment (tig) [reply Terminal.Gui/Drawing/LineCanvas.cs:319]: And it looks like ThickDotted has same issue (horizontal is not getting the right glyph)

**Review by tig — CHANGES_REQUESTED** [review]
Looking great. Big add.

See the bugs I think I found in my comments.

**Comment by BDisp** [comment]
Same as https://github.com/gui-cs/Terminal.Gui/pull/2542#issuecomment-1508510240. The `LineCanvas` is bad clipped.

**Review by tig — APPROVED** [review]

File: Terminal.Gui/Drawing/LineCanvas.cs (hunk 1 above)
Comment (tig) [reply Terminal.Gui/Drawing/LineCanvas.cs:319]: > The ~padding~ border in this image isn't rendering well.
> 
> **Edit:** Sorry, my bad. Is the border limits that isn't rendering well. The border frame is correct but not the LineCanvas rendering.

This is intentional for now; I want clipping turned off until we are sure everyone is drawing in the right places.

Comment (BDisp) [reply Terminal.Gui/Drawing/LineCanvas.cs:319]: > This is intentional for now; I want clipping turned off until we are sure everyone is drawing in the right places.

Yes I know, but I'm talking about this:

imagem

The border `LineCanvas` should be inside the border frame and it isn't.

Comment (tig) [reply Terminal.Gui/Drawing/LineCanvas.cs:319]: I see. Yep. That's a bug.

**Comment by tig** [comment]
here's the fix:

```cs
		var borderBounds = new Rect (
				screenBounds.X + Math.Max (0, Thickness.Left - 1),
				screenBounds.Y + Math.Max (0, Thickness.Top - 1),
				screenBounds.Width - (Math.Max (0, Math.Max (0, Thickness.Left - 1) + Math.Max (0, Thickness.Right - 1))),
				screenBounds.Height - (Math.Max (0, Math.Max (0, Thickness.Top - 1) + Math.Max (0, Thickness.Bottom - 1))));
```

**Comment by BDisp** [comment]
> here's the fix:
> 
> ```cs
> 		var borderBounds = new Rect (
> 				screenBounds.X + Math.Max (0, Thickness.Left - 1),
> 				screenBounds.Y + Math.Max (0, Thickness.Top - 1),
> 				screenBounds.Width - (Math.Max (0, Math.Max (0, Thickness.Left - 1) + Math.Max (0, Thickness.Right - 1))),
> 				screenBounds.Height - (Math.Max (0, Math.Max (0, Thickness.Top - 1) + Math.Max (0, Thickness.Bottom - 1))));
> ```

Thanks @tig. That throws exceptions if dimensions are negative. This is more accurate:

```cs
			var borderBounds = new Rect (
				screenBounds.X + Math.Max (0, Thickness.Left - 1),
				screenBounds.Y + Math.Max (0, Thickness.Top - 1),
				Math.Max (0, screenBounds.Width - Math.Max (0, Math.Max (0, Thickness.Left - 1) + Math.Max (0, Thickness.Right - 1))),
				Math.Max (0, screenBounds.Height - Math.Max (0, Math.Max (0, Thickness.Top - 1) + Math.Max (0, Thickness.Bottom - 1))));

```

**Comment by Nutzzz** [comment]
FWIW, here were some experiments with one more dash type.  Unfortunately, the trick I used to keep horizontal and vertical looking somewhat similar won't work without mixing two different glyphs for the horizontal line.
image
image

**Comment by Nutzzz** [comment]
It looks like there was a bad merge on the unit tests, but now there are failures.  I'm investigating...

**Comment by tig** [comment]
> > here's the fix:
> 
> > 
> 
> > ```cs
> 
> > 		var borderBounds = new Rect (
> 
> > 				screenBounds.X + Math.Max (0, Thickness.Left - 1),
> 
> > 				screenBounds.Y + Math.Max (0, Thickness.Top - 1),
> 
> > 				screenBounds.Width - (Math.Max (0, Math.Max (0, Thickness.Left - 1) + Math.Max (0, Thickness.Right - 1))),
> 
> > 				screenBounds.Height - (Math.Max (0, Math.Max (0, Thickness.Top - 1) + Math.Max (0, Thickness.Bottom - 1))));
> 
> > ```
> 
> 
> 
> Thanks @tig. That throws exceptions if dimensions are negative. This is more accurate:
> 
> 
> 
> ```cs
> 
> 			var borderBounds = new Rect (
> 
> 				screenBounds.X + Math.Max (0, Thickness.Left - 1),
> 
> 				screenBounds.Y + Math.Max (0, Thickness.Top - 1),
> 
> 				Math.Max (0, screenBounds.Width - Math.Max (0, Math.Max (0, Thickness.Left - 1) + Math.Max (0, Thickness.Right - 1))),
> 
> 				Math.Max (0, screenBounds.Height - Math.Max (0, Math.Max (0, Thickness.Top - 1) + Math.Max (0, Thickness.Bottom - 1))));
> 
> 
> 
> ```

Thanks. 

Reminder: all this is temporary. My dream is all these border/title things are eventually done via subviews of the Frame.

**Review by tig — APPROVED** [review]

**Comment by tig** [comment]
@Nutzzz in case you didn't see this has merge conflicts.

**Pull request merged**